internal/manifests/queryfrontend/query_frontend.go

```diff
@@ -53,13 +57,21 @@ func BuildQueryFrontend(params manifestutils.Params) ([]client.Object, error) {
 		}
 	}
 
-	svcs := services(tempo)
-
-	var manifests []client.Object
 	manifests = append(manifests, d)
+
+	svcs := services(tempo)
 	for _, s := range svcs {
 		manifests = append(manifests, s)
 	}
+
+	if tempo.Spec.Components.QueryFrontend.JaegerQuery.Ingress != nil && tempo.Spec.Components.QueryFrontend.JaegerQuery.Ingress.Enabled {
+		manifests = append(manifests, ingress(tempo))
+	}
+
+	if tempo.Spec.Components.QueryFrontend.JaegerQuery.Route != nil && tempo.Spec.Components.QueryFrontend.JaegerQuery.Route.Enabled {
+		manifests = append(manifests, route(tempo))
+	}
+
```

Comment:
Just to understand: do we allow to enable `Ingress` and `OpenShift Routes` at the same time?
* If yes: why? I don't see the value of that and can cause some confusion
* If not: should that be validated in `validateQueryFrontend`?

Reply:
+1, good idea to validate it in the webhook. I'll update the PR.

Reply:
Configuring both at the same time is not possible anymore with the updated spec (`type: route|ingress`).